pkg/api/norman/customization/gke/handler.go

```diff
@@ -0,0 +1,185 @@
+	return &handler{
+		schemas:       scaledContext.Schemas,
+		secretsLister: scaledContext.Core.Secrets(namespace.GlobalNamespace).Controller().Lister(),
+	}
+}
+
+func (h *handler) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
+
+	writer.Header().Set("Content-Type", "application/json")
+
+	capa := &Capabilities{}
+
+	if credID := req.URL.Query().Get("cloudCredentialId"); credID != "" {
+		if err := h.getCloudCredentials(req, capa, credID); err != nil {
+			return
+		}
+
+	} else if req.Method == http.MethodPost {
+		err := h.getCredentialsFromBody(writer, req, capa)
+		if err != nil {
+			return
+		}
+
+		if capa.Region == "" {
+			util.ReturnHTTPError(writer, req, httperror.InvalidBodyContent.Status, "cannot find Region name")
+		}
+	} else {
+		util.ReturnHTTPError(writer, req, httperror.Unauthorized.Status, "cannot access Google API without credentials to authenticate")
+	}
+
+	var serialized []byte
+	var errCode int
+	var err error
+
+	resourceType := mux.Vars(req)["resource"]
+
+	switch resourceType {
+	case "gkeMachineTypes":
+		if serialized, errCode, err = listMachineTypes(req.Context(), capa); err != nil {
+			logrus.Debugf("[gke-handler] error getting machine types: %v", err)
+			util.ReturnHTTPError(writer, req, errCode, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "gkeNetworks":
+		if serialized, errCode, err = listNetworks(req.Context(), capa); err != nil {
+			logrus.Debugf("[gke-handler] error getting networks: %v", err)
+			util.ReturnHTTPError(writer, req, errCode, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "gkeServiceAccounts":
+		if serialized, errCode, err = listServiceAccounts(req.Context(), capa); err != nil {
+			logrus.Debugf("[gke-handler] error getting serviceaccounts: %v", err)
+			util.ReturnHTTPError(writer, req, errCode, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "gkeSubnetworks":
+		if serialized, errCode, err = listSubnetworks(req.Context(), capa); err != nil {
+			logrus.Debugf("[gke-handler] error getting subnetworks: %v", err)
+			util.ReturnHTTPError(writer, req, errCode, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "gkeVersions":
+		if serialized, errCode, err = listVersions(req.Context(), capa); err != nil {
+			logrus.Debugf("[gke-handler] error getting versions: %v", err)
+			util.ReturnHTTPError(writer, req, errCode, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "gkeZones":
+		if serialized, errCode, err = listZones(req.Context(), capa); err != nil {
+			logrus.Debugf("[gke-handler] error getting zones: %v", err)
+			util.ReturnHTTPError(writer, req, errCode, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	default:
+		util.ReturnHTTPError(writer, req, httperror.NotFound.Status, "invalid endpoint "+resourceType)
+	}
+}
+
+func (h *handler) getCloudCredentials(req *http.Request, cap *Capabilities, credID string) error {
```

Comment:
```suggestion
func (h *handler) getCloudCredential(req *http.Request, cap *Capabilities, credID string) error {
```

Reply:
I renamed it, question about `getCredentialsFromBody` function name. Should it stay as it is or also rename ?